js/util/urlParser.js

```diff
@@ -25,9 +28,8 @@ var urlParser = {
     if (url.indexOf(' ') === -1 && url.indexOf('.') > 0) {
```

Comment:
The meaning of `isURLMissingProtocol` is sort of confusing at this point. Originally, the intent was to determine "could this be a URL, if you added a protocol to it?". So "google.com" would return true, but "google" would return false.

Since `validateDomain` is also used to determine whether a string is a valid URL, I think this function should call validateDomain before returning. And then inside parse, you could remove validateDomain from this line:
```
     if (urlParser.isURLMissingProtocol(url) && urlParser.validateDomain(urlParser.getDomain(url))) {
```

Reply:
Wouldn't that confuse more the purpose of `isURLMissingProtocol`? Also, `validateDomain` is used to determine if the domain contained in a URL is valid, that's why I think that keeping those two in separated procedures is important.

To be honest, I think that `isURLMissingProtocol` should be resumed to this:
```js
  isURLMissingProtocol: function (url) {
    return !urlParser.protocolRegex.test(url)
  }
```
As everything that wasn't validated before it should fall into the search spectrum.